File: Data Structures & Algorithms/trapping-rain-water/submission-1.cs
public class Solution {
    public int Trap(int[] height) 
    {
       //if(height == null || height.Length == 0) return 0; 
        int n = height.Length;
        int res = 0; 

        for (int i = 0; i < n; i++)
        {
            int lefth = height[i];
            int righth = height[i];

            for (int j = 0; j < i; j++)
            {
                lefth = Math.Max(lefth, height[j]);
            }
            for (int k= i+1; k < n; k++)
            {
                righth = Math.Max(righth, height[k]);
            }
            res += (Math.Min(lefth,righth)) - height[i];
        }


        return res; 
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res = 0
        l = 0
        contains = set()
        for i in range(len(s)):
            if s[i] not in contains:
                contains.add(s[i])
                res = max(res, i - l + 1)
            else:
                while s[i] in contains:
                    contains.remove(s[l])
                    l+=1
                contains.add(s[i])
        return res